### PR TITLE
:recycle: Consolidate HIGHLIGHTS.md into CHANGES.md :rocket: section

### LINKED ISSUE #11530: Consolidate HIGHLIGHTS.md into CHANGES.md :rocket: section
### Description

The "Check for updates" feature fetches a standalone `HIGHLIGHTS.md` file to show self-hosted users what they are missing in newer versions. This file duplicates information that already belongs in `CHANGES.md`, which has a `### :rocket: Epics and highlights` subsection in many version sections.

Maintaining two separate files per release is wasteful and error-prone. `HIGHLIGHTS.md` entries lack issue/PR references (the changelog format requires them), and the two files can drift out of sync.

### Use case

Consolidating into a single source of truth (`CHANGES.md`) means:
- Every highlight entry carries issue/PR references automatically (enforced by changelog format)
- One less file to maintain per release
- The `update-changelog` skill can proactively populate the `:rocket:` section during release workflows
- Self-hosted users see highlights with proper references, making it easier to explore the details

### Affected versions

2.15.0 through 2.17.x (all versions with highlights)

### PR DESCRIPTION
**Pull request by niwinz**
**Note:** This PR was created with AI assistance.

## What

- Eliminate the redundant `HIGHLIGHTS.md` file and make `CHANGES.md` the single source of truth for version highlights
- Rewrite the frontend parser to extract highlights from `### :rocket: Epics and highlights` subsections within `CHANGES.md`
- Decouple `parse-latest-released-version` from highlights extraction so it works independently of `:rocket:` content
- Conditionally render the highlights section in the "update available" modal only when non-empty
- Add missing `:rocket:` entries for 2.15.0 (MCP server) and 2.17.0 (blur, WebGL guides, MCP status, tokens visibility)
- Update the `update-changelog` skill to proactively propose highlights during release workflows, with two new anomaly types in the report script

## Why

`HIGHLIGHTS.md` duplicated information that already belongs in `CHANGES.md`. The highlights file lacked issue/PR references (the changelog format requires them), and maintaining two separate files per release was wasteful and error-prone. Consolidating into one file ensures every highlight entry carries proper references and the `update-changelog` skill can populate the section during release workflows.

## How

- **Content migration:** Added `:rocket:` section for 2.15.0 and 4 missing entries to 2.17.0 in `CHANGES.md`
- **Frontend parser rewrite:** Changed URL from `HIGHLIGHTS.md` to `CHANGES.md`, rewrote `parse-highlights` to navigate nested headings (`##` versions → `### :rocket:` subsections), rewrote `parse-latest-released-version` to find the first non-Unreleased version directly
- **Modal conditional rendering:** Wrapped highlights title and scroll section in `(when (seq highlights) ...)` to avoid empty headings
- **Skill update:** Added step 8b for proposing highlights with release notes hints, extended anomaly report with Type C (missing `:rocket:` section) and Type D (missing references)
- **Tests:** 12 tests, 24 assertions covering all edge cases (empty/missing `:rocket:`, Unreleased, multi-version extraction, nil/non-string input)

Closes #11530

### CODE REVIEW

File: .agents/skills/update-changelog/SKILL.md [hunk 1]
```diff
@@ -653,6 +690,35 @@ for pr_num in sorted(changelog_prs):
...
+    rocket_match = rocket_heading_re.search(vs)
+    if not rocket_match: continue
+    # Extract the :rocket: subsection body (up to next ### or ##)
+    rocket_body = vs[rocket_match.end():]
+    rocket_body = re.split(r'(?m)^#{2,3}\s', rocket_body)[0]
+    for line in rocket_body.splitlines():
+        line = line.strip()
+        if line.startswith('- ') and not issue_ref_re.search(line):
+            anomalies_d.append({'version': ver, 'line': line[:100]})
+
```
Comment (pabloalba) [inline .agents/skills/update-changelog/SKILL.md:728]: Type C/D — scans all of history, including patches

Type C flags every released ## X.Y.Z with no ### :rocket: heading. CHANGES.md has on the order of 100 version headings and 14 rocket headings. Patch lines such as 2.17.1 / 2.17.2 have no rocket section on purpose. Type C will list most of the file.

Type D only accepts penpot/penpot/issues links. It will flag 2.0.0 Taiga epics, 2.6.0 Design Tokens, and 2.3.0 **New plugin system.**.

The report is generated for one milestone. Flooding it with 1.x/2.x noise hides Type A/B mismatches. Step 8b plus “every released version should have at least one highlight” can also send an agent to invent rocket sections for patches and ancient versions.

Scope both checks to the current milestone section (or to x.y.0 from 2.15.0 onward, when this UI exists). Empty rocket headings (2.9.0, 2.4.0, 2.1.0) still would not count as Type C today, so if you keep a “must have at least one bullet” rule, test for items, not only the heading.

File: .opencode/skills/update-changelog/SKILL.md [hunk 2]
```diff
@@ -664,21 +730,27 @@ with open(OUTPUT, 'w') as f:
...
 
     f.write('## Summary\n\n')
     f.write(f'- **Issue in {MILESTONE}, referenced PR in different milestone or no milestone:** {n_a}\n')
     f.write(f'- **PR in {MILESTONE}, closing issue in a different milestone:** {n_b}\n')
-    f.write(f'- **Total anomalies:** {n_a + n_b}\n\n')
+    f.write(f'- **Released version missing :rocket: section:** {n_c}\n')
+    f.write(f'- **:rocket: entry without issue/PR references:** {n_d}\n')
+    f.write(f'- **Total anomalies:** {n_a + n_b + n_c + n_d}\n\n')
 
     # --- Anomalies section ---
```
Comment (pabloalba) [inline .opencode/skills/update-changelog/SKILL.md]: Key Principles still say “Anomaly = milestone mismatch only”

The end of the skill still describes two anomaly types. Step 11 now lists four. An agent that follows Key Principles can ignore Type C/D. Update that paragraph to match step 11.

File: .agents/skills/update-changelog/SKILL.md [hunk 3]
```diff
@@ -653,6 +690,35 @@ for pr_num in sorted(changelog_prs):
...
+    m = re.match(r'^## (\d+\.\d+\.\d+)(.*)', vs)
+    if not m: continue
+    ver, suffix = m.group(1), m.group(2)
+    if 'unreleased' in suffix.lower(): continue
+    if not rocket_heading_re.search(vs):
+        anomalies_c.append(ver)
+
+# --- Type D: :rocket: entries without issue/PR references ---
+anomalies_d = []  # list of dicts: {version, line}
+issue_ref_re = re.compile(r'\[#\d+\]\(https://github\.com/penpot/penpot/issues/\d+\)')
```
Comment (pabloalba) [inline .agents/skills/update-changelog/SKILL.md:713]: Type D does not match its own spec

The skill requires [#ISSUE] and (PR: [#PR]). The regex only checks for an issue URL. 2.16.0’s PR:[9113](...) (missing # and space) and the second 2.16.0 bullet (issue only) both pass. Either check PR format too, or document that issue-only lines are allowed (that matches changelog pairing rules).
